python/rotate_array.py

```diff
@@ -0,0 +1,27 @@
+class Solution:
+    def rotate(self, nums: List[int], k: int) -> None:
+        """
+        Link to problem: https://leetcode.com/problems/rotate-array/
+        - Given an array, rotate the array to the right by k steps,
+          where k is non-negative.
+        - Do not return anything, modify nums in-place instead.
+
+        Time complexity: O(n)
+        Space complexity: O(n)
```

Comment:
Do you know how to do it in-place?

Reply:
No... tips?

Reply:
Hm I might be wrong but I'm thinking something like this?
```
for index, num in enumerate(nums):
    temp = nums[index+k]
    nums[index+k] = nums[index]
    nums[index] = previous
    previous = temp # previous will be null value at first but will be replaced
    # iteration should be stopped once all values are replaced, in my test case I iterated k times. I'm not sure if that's going to be the case for all
    # Needs to take care of array overflow by looping back using %
```

What do you think?

Reply:
Yeah, I think it should work. I will give it a try :) Thanks